lumibot/data_sources/data_source_backtesting.py

```diff
@@ -1,17 +1,23 @@
-import logging
+from abc import ABC
 from datetime import datetime, timedelta
 
-import pandas as pd
 from lumibot.data_sources import DataSource
-from lumibot.tools import print_progress_bar, to_datetime_aware
+from lumibot.tools import print_progress_bar
 
 
-class DataSourceBacktesting(DataSource):
+class DataSourceBacktesting(DataSource, ABC):
+    """
+    This class is the base class for all backtesting data sources.  It is also an abstract class and should not be
+    instantiated directly because it does not define all necessary methods. Instead, instantiate one of the
+    child classes like PandasData.
+    """
     IS_BACKTESTING_DATA_SOURCE = True
 
     def __init__(
-        self, datetime_start, datetime_end, backtesting_started=None
+        self, datetime_start, datetime_end, backtesting_started=None, config=None, api_key=None
```

Comment:
Why does a Backtesting data source need an api key? This only really applies to polygon, and maybe new data sources we add in the future

Reply:
Seems like it should be possible to use Alpaca/Yahoo/IBKR/Tradier/TDAmeriTrade/etc. as backtest sources and an api_key is a common item across many of them.  This is to mainly to future proof the design structure so that it is easy to plug in new things will minimal changes.  This means that you don't need to add support for "polygon_api_key" and "tradier_api_key" and "alpaca_api_key", you can just have a single easy specifier for all of them.  If some DataSource doesn't need the api_key, they can just ignore it (i.e. IBKR).

Reply:
sounds good. just need to make sure our existing strategies are backwards compatible